MiniSpace.Services.Reactions/src/MiniSpace.Services.Reactions.Application/Commands/Handlers/CreateReactionHandler.cs

```diff
@@ -0,0 +1,73 @@
+using Convey.CQRS.Commands;
+using MiniSpace.Services.Reactions.Application.Events;
+using MiniSpace.Services.Reactions.Application.Exceptions;
+using MiniSpace.Services.Reactions.Application.Services;
+using MiniSpace.Services.Reactions.Core.Entities;
+using MiniSpace.Services.Reactions.Core.Exceptions;
+using MiniSpace.Services.Reactions.Core.Repositories;
+
+namespace MiniSpace.Services.Reactions.Application.Commands.Handlers
+{
+    public class CreateReactionHandler(IReactionRepository reactionRepository,
+                                 IStudentRepository studentRepository,
+                                 IPostRepository postRepository,
+                                 IEventRepository eventRepository,
+                                 IDateTimeProvider dateTimeProvider,
+                                 IMessageBroker messageBroker) : ICommandHandler<CreateReaction>
+    {
+        private readonly IReactionRepository _reactionRepository = reactionRepository;
+        private readonly IStudentRepository _studentRepository = studentRepository;
+        private readonly IPostRepository _postRepository = postRepository;
+        private readonly IEventRepository _eventRepository = eventRepository;
+        private readonly IDateTimeProvider _dateTimeProvider = dateTimeProvider;
+        private readonly IMessageBroker _messageBroker = messageBroker;
+
+        public async Task HandleAsync(CreateReaction command, CancellationToken cancellationToken = default)
+        {
+            // Check if student exists
+            if (!await _studentRepository.ExistsAsync(command.StudentId))
```

Comment:
There should be some simple authorization if student id from request matches the one from Identity